k8s-clouddriver-plugin/src/main/kotlin/com/amazon/spinnaker/clouddriver/k8s/services/GitRepoCredentials.kt

```diff
@@ -0,0 +1,94 @@
+package com.amazon.spinnaker.clouddriver.k8s.services
+
+import com.amazon.spinnaker.clouddriver.k8s.configuration.PluginConfig
+import com.amazon.spinnaker.clouddriver.k8s.exceptions.ResourceNotFound
+import com.amazon.spinnaker.clouddriver.k8s.model.GitRepo
+import com.netflix.spinnaker.clouddriver.artifacts.ArtifactCredentialsRepository
+import com.netflix.spinnaker.clouddriver.artifacts.gitRepo.GitJobExecutor
+import com.netflix.spinnaker.clouddriver.artifacts.gitRepo.GitRepoArtifactCredentials
+import com.netflix.spinnaker.kork.exceptions.MissingCredentialsException
+import org.slf4j.LoggerFactory
+import java.io.File
+
+class GitRepoCredentials(
+    private val artifactCredentialsRepository: ArtifactCredentialsRepository,
+    private val pluginConfig: PluginConfig
+) {
+
+    private val executorField = GitRepoArtifactCredentials::class.java.getDeclaredField("executor")
+    private val logger = LoggerFactory.getLogger(GitRepoCredentials::class.java)
+
+    init {
+        executorField.isAccessible = true
+    }
+
+    fun getAllCredentials(): List<GitRepo> {
+        val creds = artifactCredentialsRepository.allCredentials
+        return creds.mapNotNull {
+            if (it !is GitRepoArtifactCredentials) {
+                null
```

Comment:
what's the rest output when we return `null` here? empty list?

Reply:
Yep. The default value of empty list is returned.